src/textual/widgets/_data_table.py

```diff
@@ -655,6 +657,84 @@ def get_cell_at(self, coordinate: Coordinate) -> CellType:
         row_key, column_key = self.coordinate_to_cell_key(coordinate)
         return self.get_cell(row_key, column_key)
 
+    def get_row(self, row_key: RowKey | str) -> list[CellType]:
+        """Get the values from the row identified by the given row key.
+
+        Args:
+            row_key: The key of the row.
+
+        Returns:
+            A list of the values contained within the row.
+
+        Raises:
+            RowDoesNotExist: When there is no row corresponding to the key.
+        """
+        if row_key not in self._row_locations:
+            raise RowDoesNotExist(f"Row key {row_key!r} is not valid.")
+        cell_mapping: dict[ColumnKey, CellType] = self._data.get(row_key, {})
+        ordered_row: list[CellType] = []
+        for column in self.ordered_columns:
+            cell = cell_mapping[column.key]
+            ordered_row.append(cell)
+        return ordered_row
+
+    def get_row_at(self, row_index: int) -> list[CellType]:
+        """Get the values from the cells in a row at a given index. This will
+        return the values from a row based on the rows _current position_ in
+        the table.
+
+        Args:
+            row_index: The index of the row.
+
+        Returns:
+            A list of the values contained in the row.
+
+        Raises:
+            RowDoesNotExist: If there is no row with the given index.
+        """
+        if not self.is_valid_row_index(row_index):
+            raise RowDoesNotExist(f"Row index {row_index!r} is not valid.")
+        row_key = self._row_locations.get_key(row_index)
+        return self.get_row(row_key)
+
+    def get_column(self, column_key: ColumnKey | str) -> Iterable[CellType]:
+        """Get the values from the column identified by the given column key.
+
+        Args:
+            column_key: The key of the column.
+
+        Returns:
+            A generator which yields the cells in the column.
+
+        Raises:
+            ColumnDoesNotExist: If there is no column corresponding to the key.
+        """
+        if column_key not in self._column_locations:
+            raise ColumnDoesNotExist(f"Column key {column_key!r} is not valid.")
+
+        for row_metadata in self.ordered_rows:
```

Comment:
This may be expensive. I list expression would be worthwhile, and that `self._data` could be pulled outside of the loop.